org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.xtend

```diff
@@ -60,6 +68,30 @@ class ClassFileCache {
 	
 	def void clear() {
 		cache.clear()
+		resourcesToCompile.clear()
```

Comment:
@szarnekow why is allTopLevelTypes not cleared here

Reply:
when will it be cleared?

Reply:
i see only the popCompileResult
so besides that there is gc only if i get it correct?

Reply:
I'll enhance `clear()` to clear the other maps, too.